SAMLPortal/Controllers/SetupController.cs

```diff
@@ -0,0 +1,141 @@
+											+ ";password=" + model.MySQLPassword;
+
+					MySqlConnection connection = new MySqlConnection(connectionString);
+					connection.Open();
+
+					// If everything goes well
+					var fileName = GlobalSettings.Get("CONFIG_FILE");
+					Helpers.ReplaceEnvVariableInFile(fileName, "SP_MYSQL_HOST", model.MySQLHost);
+					Helpers.ReplaceEnvVariableInFile(fileName, "SP_MYSQL_PORT", model.MySQLPort.ToString());
+					Helpers.ReplaceEnvVariableInFile(fileName, "SP_MYSQL_DB", model.MySQLDatabaseName);
+					Helpers.ReplaceEnvVariableInFile(fileName, "SP_MYSQL_USER", model.MySQLUser);
+					Helpers.ReplaceEnvVariableInFile(fileName, "SP_MYSQL_PASS", model.MySQLPassword);
+
+					GlobalSettings.InitSettingsFromEnvironment();
+
+					SAMLPortalContext context = new SAMLPortalContext();
+					context.Database.Migrate();
+
+					// MySQL Step is complete
+					Helpers.ReplaceEnvVariableInFile(fileName, "SP_CONFIG_SETUPASSISTANT_STEP", "2");
+					return Redirect("2");
+				}
+				catch (MySqlException ex)
+				{
+					Console.WriteLine(ex.Message);
+					ModelState.AddModelError(string.Empty, ex.Message);
+				}
+				catch (Exception)
+				{
+					ModelState.AddModelError(string.Empty, "An unknown error occured. Please try again.");
+				}
+			}
+
+			return View(model);
+		}
+
+		[HttpGet]
+		[Route("2")]
+		public IActionResult SecondStep()
+		{
+			SecondStepModel model = new SecondStepModel();
+
+			var countries = new CountryHelper().GetCountryData();
+			model.CountryList = new Microsoft.AspNetCore.Mvc.Rendering.SelectList(countries, "CountryShortCode", "CountryName");
+
+			return View(model);
+		}
+
+		[HttpPost]
+		[Route("2")]
+		public IActionResult SecondStep(SecondStepModel model)
+		{
+			var countries = new CountryHelper().GetCountryData();
+
+			if (ModelState.IsValid)
+			{
+				var country = countries.Where(c => c.CountryShortCode == model.CountryCode);
+				if (country.Any())
+				{
+					GlobalSettings.Store("CONFIG_CompanyName", model.CompanyName);
+					GlobalSettings.Store("CONFIG_CompanySubject", model.CompanyName);
+					GlobalSettings.Store("CONFIG_CompanyCountryCode", country.First().CountryShortCode);
+
+					Helpers.ReplaceEnvVariableInFile(GlobalSettings.Get("CONFIG_FILE"), "SP_CONFIG_SETUPASSISTANT_STEP", "3");
+
+					return Redirect("3");
+				}
+				else
+				{
+					ModelState.AddModelError(string.Empty, "Unknown country");
+				}
+
+			}
+
+			model.CountryList = new Microsoft.AspNetCore.Mvc.Rendering.SelectList(countries, "CountryShortCode", "CountryName");
+
+			return View(model);
+		}
+	}
+
+	public class SetupAsyncActionFilter : IAsyncActionFilter
+	{
+		public async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+		{
+			if (GlobalSettings.GetInt("CONFIG_SETUPASSISTANT_STEP") > 5)
+			{
+				// Setup has been completed and cannot be accessed anymore
+				context.HttpContext.Response.Redirect("/");
+			}
+
+			var resultContext = await next();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'resultContext'.](https://app.codacy.com/manual/zegorax/SAMLPortal/pullRequest?prid=5343548)